clientv3/watch.go

```diff
@@ -615,11 +615,17 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 					// send first creation event only if requested
 					if ws.initReq.createdNotify {
 						ws.outc <- *wr
+						if ws.initReq.rev == 0 {
```

Comment:
add some comments here? (when resume for a disconnection, we start watcher from a known previous revision not from the current revision, so do not update it to header.revision)